api/v1/highlights/PKPHighlightsHandler.php

```diff
@@ -0,0 +1,301 @@
+                    'handler' => [$this, 'add'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+                ],
+            ],
+            'PUT' => [
+                [
+                    'pattern' => $this->getEndpointPattern() . '/{highlightId:\d+}',
+                    'handler' => [$this, 'edit'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+                ],
+                [
+                    'pattern' => $this->getEndpointPattern() . '/order',
+                    'handler' => [$this, 'order'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+                ],
+            ],
+            'DELETE' => [
+                [
+                    'pattern' => $this->getEndpointPattern() . '/{highlightId:\d+}',
+                    'handler' => [$this, 'delete'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+                ],
+            ],
+        ];
+        parent::__construct();
+    }
+
+    /**
+     * @copydoc PKPHandler::authorize
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        if (!$request->getContext()) {
+            $roleAssignments = $this->getSiteRoleAssignments($roleAssignments);
+        }
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+
+        $rolePolicy = new PolicySet(PolicySet::COMBINING_PERMIT_OVERRIDES);
+
+        foreach ($roleAssignments as $role => $operations) {
+            $rolePolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, $role, $operations));
+        }
+        $this->addPolicy($rolePolicy);
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * Get a single highlight
+     */
+    public function get(SlimRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $highlight = Repo::highlight()->get((int) $args['highlightId'], $this->getRequest()->getContext());
+
+        if (!$highlight) {
+            return $response->withStatus(404)->withJsonError('api.highlights.404.highlightNotFound');
+        }
+
+        return $response->withJson(
+            Repo::highlight()
+                ->getSchemaMap()
+                ->map($highlight)
+            , 200
+        );
+    }
+
+    /**
+     * Get a collection of highlights
+     */
+    public function getMany(SlimRequest $slimRequest, APIResponse $response, array $args): APIResponse
+    {
+        $collector = Repo::highlight()->getCollector()
+            ->limit(self::MAX_COUNT)
+            ->offset(0);
+
+        if ($this->getRequest()->getContext()) {
+            $collector->filterByContextIds([$this->getRequest()->getContext()->getId()]);
+        } else {
+            $collector->withSiteHighlights(Collector::SITE_ONLY);
+        }
+
+        Hook::call('API::highlights::params', [$collector, $slimRequest]);
```

Comment:
`Hook::run` is preferred to `Hook::call` for new hooks; see https://github.com/pkp/pkp-lib/discussions/8089. (Here and elsewhere)